Source/NSKeyValueObserving.m

```diff
@@ -1713,10 +1713,11 @@ - (void) willChangeValueForKey: (NSString*)aKey
   pathInfo = [info lockReturningPathInfoForKey: aKey];
   if (pathInfo != nil)
     {
+      RETAIN(pathInfo);
```

Comment:
This could be the actual solution for the problem I had in NSArrayController where I have this horrible workaround that causes a memory leak.
On the other hand I remember trying something similar at that time and running into issues with that solution. Could you please give this a thorough try before merging.

Reply:
Will do.  I will add a test for this.